samples/cpp/benchmark_app/utils.cpp

```diff
@@ -122,31 +121,51 @@ std::vector<std::string> parse_devices(const std::string& device_string) {
         return std::vector<std::string>();
 
     auto devices = split(comma_separated_devices, ',');
-    return devices;
+    result.insert(result.end(), devices.begin(), devices.end());
+    return result;
 }
 
 std::map<std::string, std::string> parse_value_per_device(const std::vector<std::string>& devices,
                                                           const std::string& values_string) {
     //  Format: <device1>:<value1>,<device2>:<value2> or just <value>
     std::map<std::string, std::string> result;
     auto device_value_strings = split(values_string, ',');
+    auto if_auto = std::find(devices.begin(), devices.end(), "AUTO") != devices.end();
+    auto if_multi = std::find(devices.begin(), devices.end(), "MULTI") != devices.end();
     for (auto& device_value_string : device_value_strings) {
         auto device_value_vec = split(device_value_string, ':');
         if (device_value_vec.size() == 2) {
             auto device_name = device_value_vec.at(0);
             auto nstreams = device_value_vec.at(1);
             auto it = std::find(devices.begin(), devices.end(), device_name);
             if (it != devices.end()) {
-                result[device_name] = nstreams;
+                if (!if_auto && !if_multi) {
+                    result[device_name] = nstreams;
```

Comment:
Plese note, that this function is used not only for nstreams parsing, but for inference precision also

Reply:
sure.